crates/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -428,6 +428,15 @@ impl Context {
                         // assert_eq!(result_ids.len(), outputs.len());
 
                         for (result, output) in result_ids.iter().zip(outputs) {
+                            // when the intrinsic returns an array, we initialize it
+                            if let AcirValue::Array(elements) = &output {
```

Comment:
Should we panic if we get a dynamic array to make sure that we never have a dynamic array output from `convert_ssa_intrinsic_call`?

Reply:
There is unreachable now in case of dynamic array.